File: ATLASDEV/TTObjectClasses/ExternalWebServices/MedulaWebServiceDefinition/Acil112Servis/Acil112Servis.cs

using System;
using System.Xml;
using System.Data;
using System.Text;
using System.Drawing;
using System.Reflection;
using System.Collections;
using System.Linq;
using System.ComponentModel;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Collections.ObjectModel;
using System.Runtime.InteropServices;

using TTUtils;
using TTObjectClasses;
using TTDataDictionary;
using TTCoreDefinitions;
using TTConnectionManager;
using TTInstanceManagement;
using TTDefinitionManagement;
using TTStorageManager.Security;



using TTStorageManager;
using System.Runtime.Versioning;


namespace TTObjectClasses
{
    public  partial class Acil112Servis : TTObject
    {
        
                    
        public static partial class WebMethods
        {
                    
        }
                    
#region Methods
        /// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "2.0.50727.5420")]
    [System.SerializableAttribute()]
    [System.Diagnostics.DebuggerStepThroughAttribute()]
    [System.ComponentModel.DesignerCategoryAttribute("code")]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="http://tempuri.org/")]
    public partial class VakaBilgileri {
        
        private int protokolNoField;
        
        private string hastaAdiSoyadiField;
        
        private string vakaSenesiField;
        
        private string kimlikNoField;
        
        /// <remarks/>
        public int protokolNo {
            get {
                return protokolNoField;
            }
            set {
                protokolNoField = value;
            }
        }
        
        /// <remarks/>
        public string hastaAdiSoyadi {
            get {
                return hastaAdiSoyadiField;
            }
            set {
                hastaAdiSoyadiField = value;
            }
        }
        
        /// <remarks/>
        public string vakaSenesi {
            get {
                return vakaSenesiField;
            }
            set {
                vakaSenesiField = value;
            }
        }
        
        /// <remarks/>
        public string kimlikNo {
            get {
                return kimlikNoField;
            }
            set {
                kimlikNoField = value;
            }
        }
    }
    
    /// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "2.0.50727.5420")]
    [System.SerializableAttribute()]
    [System.Diagnostics.DebuggerStepThroughAttribute()]
    [System.ComponentModel.DesignerCategoryAttribute("code")]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="http://tempuri.org/")]
    public partial class BolumBilgileri {
        
        private string bolumAdiField;
        
        private int toplamYatakField;
        
        private int bosYatakField;
        
        private int toplamVentilField;
        
        private int bosVentilField;
        
        private int toplamKuvezField;
        
        private int bosKuvezField;
        
        /// <remarks/>
        public string BolumAdi {
            get {
                return bolumAdiField;
            }
            set {
                bolumAdiField = value;
            }
        }
        
        /// <remarks/>
        public int ToplamYatak {
            get {
                return toplamYatakField;
            }
            set {
                toplamYatakField = value;
            }
        }
        
        /// <remarks/>
        public int BosYatak {
            get {
                return bosYatakField;
            }
            set {
                bosYatakField = value;
            }
        }
        
        /// <remarks/>
        public int ToplamVentil {
            get {
                return toplamVentilField;
            }
            set {
                toplamVentilField = value;
            }
        }
        
        /// <remarks/>
        public int BosVentil {
            get {
                return bosVentilField;
            }
            set {
                bosVentilField = value;
            }
        }
        
        /// <remarks/>
        public int ToplamKuvez {
            get {
                return toplamKuvezField;
            }
            set {
                toplamKuvezField = value;
            }
        }
        
        /// <remarks/>
        public int BosKuvez {
            get {
                return bosKuvezField;
            }
            set {
                bosKuvezField = value;
            }
        }
    }
    
    /// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "2.0.50727.5420")]
    [System.SerializableAttribute()]
    [System.Diagnostics.DebuggerStepThroughAttribute()]
    [System.ComponentModel.DesignerCategoryAttribute("code")]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="http://tempuri.org/")]
    public partial class HastaBilgileri {
        
        private string adiSoyadiField;
        
        private string tCKimlikNoField;
        
        private int yasField;
        
        private string cinsiyetField;
        
        private string hastaGirisZamaniField;
        
        private bool yatarakTedaviField;
        
        private string ilField;
        
        private string ilceField;
        
        private string mahalleField;
        
        /// <remarks/>
        public string AdiSoyadi {
            get {
                return adiSoyadiField;
            }
            set {
                adiSoyadiField = value;
            }
        }
        
        /// <remarks/>
        public string TCKimlikNo {
            get {
                return tCKimlikNoField;
            }
            set {
                tCKimlikNoField = value;
            }
        }
        
        /// <remarks/>
        public int Yas {
            get {
                return yasField;
            }
            set {
                yasField = value;
            }
        }
        
        /// <remarks/>
        public string Cinsiyet {
            get {
                return cinsiyetField;
            }
            set {
                cinsiyetField = value;
            }
        }
        
        /// <remarks/>
        public string HastaGirisZamani {
            get {
                return hastaGirisZamaniField;
            }
            set {
                hastaGirisZamaniField = value;
            }
        }
        
        /// <remarks/>
        public bool YatarakTedavi {
            get {
                return yatarakTedaviField;
            }
            set {
                yatarakTedaviField = value;
            }
        }
        
        /// <remarks/>
        public string Il {
            get {
                return ilField;
            }
            set {
                ilField = value;
            }
        }
        
        /// <remarks/>
        public string Ilce {
            get {
                return ilceField;
            }
            set {
                ilceField = value;
            }
        }
        
        /// <remarks/>
        public string Mahalle {
            get {
                return mahalleField;
            }
            set {
                mahalleField = value;
            }
        }
    }
    
    /// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "2.0.50727.5420")]
    [System.SerializableAttribute()]
    [System.Diagnostics.DebuggerStepThroughAttribute()]
    [System.ComponentModel.DesignerCategoryAttribute("code")]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="http://tempuri.org/")]
    public partial class BolumBilgisi {
        
        private BolumKodlari bolumKodField;
        
        private int toplamYatakField;
        
        private int bosYatakField;
        
        private int toplamVentilField;
        
        private int bosVentilField;
        
        private int toplamKuvezField;
        
        private int bosKuvezField;
        
        /// <remarks/>
        public BolumKodlari BolumKod {
            get {
                return bolumKodField;
            }
            set {
                bolumKodField = value;
            }
        }
        
        /// <remarks/>
        public int ToplamYatak {
            get {
                return toplamYatakField;
            }
            set {
                toplamYatakField = value;
            }
        }
        
        /// <remarks/>
        public int BosYatak {
            get {
                return bosYatakField;
            }
            set {
                bosYatakField = value;
            }
        }
        
        /// <remarks/>
        public int ToplamVentil {
            get {
                return toplamVentilField;
            }
            set {
                toplamVentilField = value;
            }
        }
        
        /// <remarks/>
        public int BosVentil {
            get {
                return bosVentilField;
            }
            set {
                bosVentilField = value;
            }
        }
        
        /// <remarks/>
        public int ToplamKuvez {
            get {
                return toplamKuvezField;
            }
            set {
                toplamKuvezField = value;
            }
        }
        
        /// <remarks/>
        public int BosKuvez {
            get {
                return bosKuvezField;
            }
            set {
                bosKuvezField = value;
            }
        }
    }
    
    /// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "2.0.50727.5420")]
    [System.SerializableAttribute()]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="http://tempuri.org/")]
    public enum BolumKodlari {
        
        /// <remarks/>
        AcilEriskin,
        
        /// <remarks/>
        AcilCocuk,
        
        /// <remarks/>
        YogunBakimKoroner,
        
        /// <remarks/>
        YogunBakimKalpDamar,
        
        /// <remarks/>
        YogunBakimAnestezi_Reanimasyon,
        
        /// <remarks/>
        YogunBakimNoroloji,
        
        /// <remarks/>
        YogunBakimBeyinCerrahisi,
        
        /// <remarks/>
        YogunBakimGenelCerrahi,
        
        /// <remarks/>
        YogunBakimDahiliye,
        
        /// <remarks/>
        YogunBakimGogusHastaliklari,
        
        /// <remarks/>
        YogunBakimKadinDogum,
        
        /// <remarks/>
        YogunBakimPediatri,
        
        /// <remarks/>
        YogunBakimYenidogan,
        
        /// <remarks/>
        YogunBakimYanik,
        
        /// <remarks/>
        ServisPediatri,
        
        /// <remarks/>
        ServisDahiliye,
        
        /// <remarks/>
        ServisKalpDamar,
        
        /// <remarks/>
        ServisKadinDogum,
        
        /// <remarks/>
        ServisGenelCerrahi,
        
        /// <remarks/>
        ServisOrtopedi,
        
        /// <remarks/>
        ServisKalpDamarCerrahisi,
        
        /// <remarks/>
        ServisBeyinCerrahisi,
        
        /// <remarks/>
        ServisYenidogan,
        
        /// <remarks/>
        ServisDializ,
        
        /// <remarks/>
        ServisDiger,
        
        /// <remarks/>
        Ameliyathane,
        
        /// <remarks/>
        Morg,
        
        /// <remarks/>
        YogunBakimCocukCerrahi,
        
        /// <remarks/>
        ServisCocukCerrahi,
        
        /// <remarks/>
        ServisCocukKardiyolojisi,
        
        /// <remarks/>
        ServisCocukHematolojiOnkoloji,
        
        /// <remarks/>
        ServisCocukKKB,
        
        /// <remarks/>
        ServisCocukGastroloji,
        
        /// <remarks/>
        ServisCocukPsikoloji,
        
        /// <remarks/>
        ServisCocukNoroloji,
        
        /// <remarks/>
        ServisCocukHastaliklari,
        
        /// <remarks/>
        ServisKardiyoloji,
        
        /// <remarks/>
        YogunBakimDiger,
        
        /// <remarks/>
        ServisEnfeksiyon,
        
        /// <remarks/>
        YogunBakimEnfeksiyon,
    }
    
    /// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "2.0.50727.5420")]
    [System.SerializableAttribute()]
    [System.Diagnostics.DebuggerStepThroughAttribute()]
    [System.ComponentModel.DesignerCategoryAttribute("code")]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="http://tempuri.org/")]
    public partial class NobetciPersonelBilgileri {
        
        private string gorevField;
        
        private string bransField;
        
        private string bolumField;
        
        private string personelSicilNoField;
        
        private string nobetTarihiBaslangicField;
        
        private string nobetTarihiBitisField;
        
        private string personelAdiField;
        
        private string personelSoyadiField;
        
        private string telefonField;
        
        private bool icapField;
        
        /// <remarks/>
        public string gorev {
            get {
                return gorevField;
            }
            set {
                gorevField = value;
            }
        }
        
        /// <remarks/>
        public string brans {
            get {
                return bransField;
            }
            set {
                bransField = value;
            }
        }
        
        /// <remarks/>
        public string bolum {
            get {
                return bolumField;
            }
            set {
                bolumField = value;
            }
        }
        
        /// <remarks/>
        public string personelSicilNo {
            get {
                return personelSicilNoField;
            }
            set {
                personelSicilNoField = value;
            }
        }
        
        /// <remarks/>
        public string nobetTarihiBaslangic {
            get {
                return nobetTarihiBaslangicField;
            }
            set {
                nobetTarihiBaslangicField = value;
            }
        }
        
        /// <remarks/>
        public string nobetTarihiBitis {
            get {
                return nobetTarihiBitisField;
            }
            set {
                nobetTarihiBitisField = value;
            }
        }
        
        /// <remarks/>
        public string personelAdi {
            get {
                return personelAdiField;
            }
            set {
                personelAdiField = value;
            }
        }
        
        /// <remarks/>
        public string personelSoyadi {
            get {
                return personelSoyadiField;
            }
            set {
                personelSoyadiField = value;
            }
        }
        
        /// <remarks/>
        public string telefon {
            get {
                return telefonField;
            }
            set {
                telefonField = value;
            }
        }
        
        /// <remarks/>
        public bool icap {
            get {
                return icapField;
            }
            set {
                icapField = value;
            }
        }
    }
    
    /// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Xml", "2.0.50727.5420")]
    [System.SerializableAttribute()]
    [System.Diagnostics.DebuggerStepThroughAttribute()]
    [System.ComponentModel.DesignerCategoryAttribute("code")]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="http://tempuri.org/")]
    public partial class PersonelBilgileri {
        
        private string gorevField;
        
        private string bransField;
        
        private string bolumField;
        
        private string personelSicilNoField;
        
        private string nobetTarihiBaslangicField;
        
        private string nobetTarihiBitisField;
        
        private string personelAdiField;
        
        private string personelSoyadiField;
        
        private string telefonField;
        
        /// <remarks/>
        public string gorev {
            get {
                return gorevField;
            }
            set {
                gorevField = value;
            }
        }
        
        /// <remarks/>
        public string brans {
            get {
                return bransField;
            }
            set {
                bransField = value;
            }
        }
        
        /// <remarks/>
        public string bolum {
            get {
                return bolumField;
            }
            set {
                bolumField = value;
            }
        }
        
        /// <remarks/>
        public string personelSicilNo {
            get {
                return personelSicilNoField;
            }
            set {
                personelSicilNoField = value;
            }
        }
        
        /// <remarks/>
        public string nobetTarihiBaslangic {
            get {
                return nobetTarihiBaslangicField;
            }
            set {
                nobetTarihiBaslangicField = value;
            }
        }
        
        /// <remarks/>
        public string nobetTarihiBitis {
            get {
                return nobetTarihiBitisField;
            }
            set {
                nobetTarihiBitisField = value;
            }
        }
        
        /// <remarks/>
        public string personelAdi {
            get {
                return personelAdiField;
            }
            set {
                personelAdiField = value;
            }
        }
        
        /// <remarks/>
        public string personelSoyadi {
            get {
                return personelSoyadiField;
            }
            set {
                personelSoyadiField = value;
            }
        }
        
        /// <remarks/>
        public string telefon {
            get {
                return telefonField;
            }
            set {
                telefonField = value;
            }
        }
    }
        
#endregion Methods

    }
}

File: AtlasWeb/Core/wwwroot/app/NebulaClient/Services/External/Acil112Servis.ts
//$4F45B2A4
import { Guid } from "NebulaClient/Mscorlib/Guid";
import { IWebMethodCallback } from "NebulaClient/Utils/IWebMethodCallback";
import { TTWebServiceUri } from "NebulaClient/Utils/TTWebServiceUri";
import { ServiceLocator } from "Fw/Services/ServiceLocator";

export namespace Acil112Servis {
    export enum BolumKodlari {
        /// <remarks/>
        AcilEriskin,
        /// <remarks/>
        AcilCocuk,
        /// <remarks/>
        YogunBakimKoroner,
        /// <remarks/>
        YogunBakimKalpDamar,
        /// <remarks/>
        YogunBakimAnestezi_Reanimasyon,
        /// <remarks/>
        YogunBakimNoroloji,
        /// <remarks/>
        YogunBakimBeyinCerrahisi,
        /// <remarks/>
        YogunBakimGenelCerrahi,
        /// <remarks/>
        YogunBakimDahiliye,
        /// <remarks/>
        YogunBakimGogusHastaliklari,
        /// <remarks/>
        YogunBakimKadinDogum,
        /// <remarks/>
        YogunBakimPediatri,
        /// <remarks/>
        YogunBakimYenidogan,
        /// <remarks/>
        YogunBakimYanik,
        /// <remarks/>
        ServisPediatri,
        /// <remarks/>
        ServisDahiliye,
        /// <remarks/>
        ServisKalpDamar,
        /// <remarks/>
        ServisKadinDogum,
        /// <remarks/>
        ServisGenelCerrahi,
        /// <remarks/>
        ServisOrtopedi,
        /// <remarks/>
        ServisKalpDamarCerrahisi,
        /// <remarks/>
        ServisBeyinCerrahisi,
        /// <remarks/>
        ServisYenidogan,
        /// <remarks/>
        ServisDializ,
        /// <remarks/>
        ServisDiger,
        /// <remarks/>
        Ameliyathane,
        /// <remarks/>
        Morg,
        /// <remarks/>
        YogunBakimCocukCerrahi,
        /// <remarks/>
        ServisCocukCerrahi,
        /// <remarks/>
        ServisCocukKardiyolojisi,
        /// <remarks/>
        ServisCocukHematolojiOnkoloji,
        /// <remarks/>
        ServisCocukKKB,
        /// <remarks/>
        ServisCocukGastroloji,
        /// <remarks/>
        ServisCocukPsikoloji,
        /// <remarks/>
        ServisCocukNoroloji,
        /// <remarks/>
        ServisCocukHastaliklari,
        /// <remarks/>
        ServisKardiyoloji,
        /// <remarks/>
        YogunBakimDiger,
        /// <remarks/>
        ServisEnfeksiyon,
        /// <remarks/>
        YogunBakimEnfeksiyon
    }

    export class VakaBilgileri {
        public protokolNo: number;
        public hastaAdiSoyadi: string;
        public vakaSenesi: string;
        public kimlikNo: string;
    }

    export class BolumBilgileri {
        public BolumAdi: string;
        public ToplamYatak: number;
        public BosYatak: number;
        public ToplamVentil: number;
        public BosVentil: number;
        public ToplamKuvez: number;
        public BosKuvez: number;
    }

    export class HastaBilgileri {
        public AdiSoyadi: string;
        public TCKimlikNo: string;
        public Yas: number;
        public Cinsiyet: string;
        public HastaGirisZamani: string;
        public YatarakTedavi: boolean;
        public Il: string;
        public Ilce: string;
        public Mahalle: string;
    }

    export class BolumBilgisi {
        public BolumKod: BolumKodlari;
        public ToplamYatak: number;
        public BosYatak: number;
        public ToplamVentil: number;
        public BosVentil: number;
        public ToplamKuvez: number;
        public BosKuvez: number;
    }

    export class NobetciPersonelBilgileri {
        public gorev: string;
        public brans: string;
        public bolum: string;
        public personelSicilNo: string;
        public nobetTarihiBaslangic: string;
        public nobetTarihiBitis: string;
        public personelAdi: string;
        public personelSoyadi: string;
        public telefon: string;
        public icap: boolean;
    }

    export class PersonelBilgileri {
        public gorev: string;
        public brans: string;
        public bolum: string;
        public personelSicilNo: string;
        public nobetTarihiBaslangic: string;
        public nobetTarihiBitis: string;
        public personelAdi: string;
        public personelSoyadi: string;
        public telefon: string;
    }

    export class WebMethods {
        public static async KayitNoProtokolNoDonusumSync(siteID: Guid, webServisUri: TTWebServiceUri, kayitNo: string, protokolNo: string, sorguYili: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/KayitNoProtokolNoDonusumSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "kayitNo": kayitNo, "protokolNo": protokolNo, "sorguYili": sorguYili };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async NobetciPersonelGonder_ArraySync(siteID: Guid, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, personelListesi: PersonelBilgileri[]): Promise<number> {
            let url: string = "/api/Acil112ServisController/NobetciPersonelGonder_ArraySync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "personelListesi": personelListesi };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async NobetciPersonelGondermeMetodu_ArraySync(siteID: Guid, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, nobetciler: NobetciPersonelBilgileri[]): Promise<number> {
            let url: string = "/api/Acil112ServisController/NobetciPersonelGondermeMetodu_ArraySync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "nobetciler": nobetciler };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async EUYSTaniGonderASync(siteID: Guid, callerObject: IWebMethodCallback, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, hasta: HastaBilgileri, ICD10Kodu: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/EUYSTaniGonderASync";
            let inputDto = { "siteID": siteID, "callerObject": callerObject, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "hasta": hasta, "ICD10Kodu": ICD10Kodu };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async VakaSonlandirmaMetoduASync(siteID: Guid, callerObject: IWebMethodCallback, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, protokolNo: string, tarih: string, hastaKabulTarihi: string, hastaAyrilisTarihi: string, vakaSonuc: string, aciklama: string, XXXXXXSonucTanisi: string, sosyalGuvenlikKurumu: string, sosyalGuvenlikNo: string, tcKimlikNo: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/VakaSonlandirmaMetoduASync";
            let inputDto = { "siteID": siteID, "callerObject": callerObject, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "protokolNo": protokolNo, "tarih": tarih, "hastaKabulTarihi": hastaKabulTarihi, "hastaAyrilisTarihi": hastaAyrilisTarihi, "vakaSonuc": vakaSonuc, "aciklama": aciklama, "XXXXXXSonucTanisi": XXXXXXSonucTanisi, "sosyalGuvenlikKurumu": sosyalGuvenlikKurumu, "sosyalGuvenlikNo": sosyalGuvenlikNo, "tcKimlikNo": tcKimlikNo };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async NobetciPersonelGondermeMetoduSync(siteID: Guid, webServisUri: TTWebServiceUri, nobetci: NobetciPersonelBilgileri): Promise<number> {
            let url: string = "/api/Acil112ServisController/NobetciPersonelGondermeMetoduSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "nobetci": nobetci };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async NobetciPersonelGonderSync(siteID: Guid, webServisUri: TTWebServiceUri, personelListesi: Object): Promise<number> {
            let url: string = "/api/Acil112ServisController/NobetciPersonelGonderSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "personelListesi": personelListesi };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async ProtokolDetaylariDonderSync(siteID: Guid, webServisUri: TTWebServiceUri, protokolNo: string, sene: string, hastaAdiSoyadi: string, TCKimlikNo: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/ProtokolDetaylariDonderSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "protokolNo": protokolNo, "sene": sene, "hastaAdiSoyadi": hastaAdiSoyadi, "TCKimlikNo": TCKimlikNo };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async ServiceTestSync(siteID: Guid, webServisUri: TTWebServiceUri): Promise<string> {
            let url: string = "/api/Acil112ServisController/ServiceTestSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri };
            let result = await ServiceLocator.post(url, inputDto) as string;
            return result;
        }
        public static async SonlandirilmamisVakalarListesiDetayliSync(siteID: Guid, webServisUri: TTWebServiceUri, acikVakalar: VakaBilgileri[]): Promise<number> {
            let url: string = "/api/Acil112ServisController/SonlandirilmamisVakalarListesiDetayliSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "acikVakalar": acikVakalar };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async SonlandirilmamisVakalarListesiSync(siteID: Guid, webServisUri: TTWebServiceUri, protokoller: number[]): Promise<number> {
            let url: string = "/api/Acil112ServisController/SonlandirilmamisVakalarListesiSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "protokoller": protokoller };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async SosyalGuvenceKodlariDonderSync(siteID: Guid, webServisUri: TTWebServiceUri, kurumKodlari: string[]): Promise<number> {
            let url: string = "/api/Acil112ServisController/SosyalGuvenceKodlariDonderSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "kurumKodlari": kurumKodlari };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async VakaGuncellemeMetoduSync(siteID: Guid, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, protokolNo: string, tarih: string, XXXXXXKabulTarihi: string, aciklama: string, ip: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/VakaGuncellemeMetoduSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "protokolNo": protokolNo, "tarih": tarih, "XXXXXXKabulTarihi": XXXXXXKabulTarihi, "aciklama": aciklama, "ip": ip };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async VakaSonlandirmaMetoduSync(siteID: Guid, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, protokolNo: string, tarih: string, hastaKabulTarihi: string, hastaAyrilisTarihi: string, vakaSonuc: string, aciklama: string, XXXXXXSonucTanisi: string, sosyalGuvenlikKurumu: string, sosyalGuvenlikNo: string, tcKimlikNo: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/VakaSonlandirmaMetoduSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "protokolNo": protokolNo, "tarih": tarih, "hastaKabulTarihi": hastaKabulTarihi, "hastaAyrilisTarihi": hastaAyrilisTarihi, "vakaSonuc": vakaSonuc, "aciklama": aciklama, "XXXXXXSonucTanisi": XXXXXXSonucTanisi, "sosyalGuvenlikKurumu": sosyalGuvenlikKurumu, "sosyalGuvenlikNo": sosyalGuvenlikNo, "tcKimlikNo": tcKimlikNo };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async AdetleriGonder_ArraySync(siteID: Guid, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, bolumler: BolumBilgisi[], ipAddr: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/AdetleriGonder_ArraySync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "bolumler": bolumler, "ipAddr": ipAddr };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async AdetleriGonderSync(siteID: Guid, webServisUri: TTWebServiceUri, acilEriskin: BolumBilgileri, acilCocuk: BolumBilgileri, ybKoroner: BolumBilgileri, ybKalpDamar: BolumBilgileri, ybAnestezi: BolumBilgileri, ybNoroloji: BolumBilgileri, ybBeyinCerrahi: BolumBilgileri, ybGenelCerrahi: BolumBilgileri, ybDahiliye: BolumBilgileri, ybGogusHastaliklari: BolumBilgileri, ybKadinDogum: BolumBilgileri, ybPediatri: BolumBilgileri, ybYenidogan: BolumBilgileri, ybYanik: BolumBilgileri, srvPediatri: BolumBilgileri, srvDahiliye: BolumBilgileri, srvKalpDamar: BolumBilgileri, srvKadinDogum: BolumBilgileri, srvGenelCerrahi: BolumBilgileri, srvOrtopedi: BolumBilgileri, srvKalpDamarCerrahi: BolumBilgileri, srvBeyinCerrahi: BolumBilgileri, srvYenidogan: BolumBilgileri, srvDializ: BolumBilgileri, srvDiger: BolumBilgileri, ameliyathane: BolumBilgileri, morg: BolumBilgileri, ipAddr: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/AdetleriGonderSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "acilEriskin": acilEriskin, "acilCocuk": acilCocuk, "ybKoroner": ybKoroner, "ybKalpDamar": ybKalpDamar, "ybAnestezi": ybAnestezi, "ybNoroloji": ybNoroloji, "ybBeyinCerrahi": ybBeyinCerrahi, "ybGenelCerrahi": ybGenelCerrahi, "ybDahiliye": ybDahiliye, "ybGogusHastaliklari": ybGogusHastaliklari, "ybKadinDogum": ybKadinDogum, "ybPediatri": ybPediatri, "ybYenidogan": ybYenidogan, "ybYanik": ybYanik, "srvPediatri": srvPediatri, "srvDahiliye": srvDahiliye, "srvKalpDamar": srvKalpDamar, "srvKadinDogum": srvKadinDogum, "srvGenelCerrahi": srvGenelCerrahi, "srvOrtopedi": srvOrtopedi, "srvKalpDamarCerrahi": srvKalpDamarCerrahi, "srvBeyinCerrahi": srvBeyinCerrahi, "srvYenidogan": srvYenidogan, "srvDializ": srvDializ, "srvDiger": srvDiger, "ameliyathane": ameliyathane, "morg": morg, "ipAddr": ipAddr };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async EUYSTaniGonderSync(siteID: Guid, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, hasta: HastaBilgileri, ICD10Kodu: string): Promise<number> {
            let url: string = "/api/Acil112ServisController/EUYSTaniGonderSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "hasta": hasta, "ICD10Kodu": ICD10Kodu };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
        public static async GetVersionNoSync(siteID: Guid, webServisUri: TTWebServiceUri): Promise<string> {
            let url: string = "/api/Acil112ServisController/GetVersionNoSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri };
            let result = await ServiceLocator.post(url, inputDto) as string;
            return result;
        }
        public static async XXXXXXPersoneliniDonderSync(siteID: Guid, webServisUri: TTWebServiceUri, userName1: string, userPassword1: string, personelListesi: PersonelBilgileri[]): Promise<number> {
            let url: string = "/api/Acil112ServisController/XXXXXXPersoneliniDonderSync";
            let inputDto = { "siteID": siteID, "webServisUri": webServisUri, "userName1": userName1, "userPassword1": userPassword1, "personelListesi": personelListesi };
            let result = await ServiceLocator.post(url, inputDto) as number;
            return result;
        }
    }
}
